backend/substrapp/views/compositetraintuple.py

```diff
@@ -0,0 +1,118 @@
+import logging
+
+from rest_framework import mixins, status
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from substrapp.serializers import LedgerCompositeTraintupleSerializer
+from substrapp.ledger_utils import query_ledger, get_object_from_ledger, LedgerError, LedgerConflict
+from substrapp.views.filters_utils import filter_list
+from substrapp.views.utils import validate_pk, get_success_create_code, LedgerException
+
+
+class CompositeTraintupleViewSet(mixins.CreateModelMixin,
+                                 mixins.RetrieveModelMixin,
+                                 mixins.ListModelMixin,
+                                 GenericViewSet):
+    serializer_class = LedgerCompositeTraintupleSerializer
+    ledger_query_call = 'queryCompositeTraintuple'
+
+    def get_queryset(self):
+        return []
+
+    def perform_create(self, serializer):
+        return serializer.save()
+
+    def commit(self, serializer, pkhash):
+        # create on ledger
+        try:
+            data = serializer.create(serializer.validated_data)
+        except LedgerError as e:
+            raise LedgerException({'message': str(e.msg), 'pkhash': pkhash}, e.status)
+        else:
+            return data
+
+    def _create(self, request):
+        data = {
+            'algo_key': request.data.get('algo_key'),
+            'data_manager_key': request.data.get('data_manager_key'),
+            'objective_key': request.data.get('objective_key'),
+            'rank': request.data.get('rank'),
+            'compute_plan_id': request.data.get('compute_plan_id', ''),
+            'in_head_model_key': request.data.get('in_head_model_key', ''),
+            'in_trunk_model_key': request.data.get('in_trunk_model_key', ''),
+            'out_trunk_model_permissions': {
+                'public': False,
```

Comment:
Are we sure we want to force the out trunk model to be private? If so shouldn't we reject input data with public trunk model permissions? Otherwise we'll endup just ignoring user input.

Reply:
The trunk should not be public but limited for a set of authorized nodes. so yes, be default public is false and only list should be set.

Reply:
Got it thanks, I'll make a comment in the related `substra` PR to update the doc accordingly.